### PR TITLE
Delete sticky notes with right-click menu

### PR DESCRIPTION
**Pull request by emgamedev-unity3d**
### Purpose of this PR
To improve the usability of Sticky Notes, added the ability to delete them through the right-click
context menu.

Note: This is my first PR ever, PLZ HELP! 😟 🙏 

---
### Release Notes
Please add any useful notes about the feature/fix that might be helpful for others.

---
### Testing status
**Katana Tests**: First off we need to make sure the Katana SRP tests are green?

**Manual Tests**: What did you do?
- [ ] Opened test project + Run graphic tests locally
- [ ] Built a player
- [ ] Checked new UI names with UX convention
- [x] Tested UI multi-edition + Undo/Redo
- [ ] C# and shader warnings (supress shader cache to see them)
- [ ] Checked new resources path for the reloader (in devloper mode, you have a button at end of resources that check the pathes)
- Other: 

**Automated Tests**: What did you setup? (Add a screenshot or the reference image of the test please)

Launch Katana (Link on master here - update for your branch):
https://katana.bf.unity3d.com/projects/com.unity.render-pipelines/builders?ScriptableRenderLoop_branch=esteban%2Festicky-node&automation-tools_branch=master&unity_branch=trunk&sort=1-asc

Alternative, launch Yamato (Select your branch):
https://yamato.prd.cds.internal.unity3d.com/jobs/78-ScriptableRenderPipeline/tree/esteban%252Festicky-node

Any test projects to go with this to help reviewers?

---
### Overall Product Risks
**Technical Risk**: None, Low, Medium, High?
Low

**Halo Effect**: None, Low, Medium, High?
Low

---
### Comments to reviewers
Notes for the reviewers you have assigned.

### CODE REVIEW

File: com.unity.shadergraph/Editor/Drawing/Views/StickyNote.cs [hunk 1]
```diff
@@ -4,6 +4,7 @@
 using UnityEngine.UIElements;
 using System.Reflection;
 using System.Linq;
+using UnityEditor.Graphs;
```
Comment (martint-unity) [inline com.unity.shadergraph/Editor/Drawing/Views/StickyNote.cs]: I don't think one is needed.

File: com.unity.shadergraph/Editor/Drawing/Views/StickyNote.cs [hunk 2]
```diff
@@ -466,9 +467,18 @@ public void BuildContextualMenu(ContextualMenuPopulateEvent evt)
 
                 evt.menu.AppendAction("Fit To Text", OnFitToText, e => DropdownMenuAction.Status.Normal);
                 evt.menu.AppendSeparator();
+                
+                evt.menu.AppendAction("Delete",  OnDestroy, e => DropdownMenuAction.Status.Normal);
+                evt.menu.AppendSeparator();
             }
         }
 
+        void OnDestroy(DropdownMenuAction menuAction)
```
Comment (martint-unity) [inline com.unity.shadergraph/Editor/Drawing/Views/StickyNote.cs]: Just nit picking here, but why name it OnDestroy and not OnDelete?

Comment (emgamedev-unity3d) [reply com.unity.shadergraph/Editor/Drawing/Views/StickyNote.cs]: true, I originally renamed this when I started and forgot about it. I'll do that change, nice catch.

File: com.unity.shadergraph/Editor/Drawing/Views/StickyNote.cs [hunk 3]
```diff
@@ -2,9 +2,6 @@
 using UnityEditor.Experimental.GraphView;
 using UnityEngine;
 using UnityEngine.UIElements;
-using System.Reflection;
-using System.Linq;
-using UnityEditor.Graphs;
```
Comment (emgamedev-unity3d) [inline com.unity.shadergraph/Editor/Drawing/Views/StickyNote.cs]: These 3 were not needed/unused

**Comment by emgamedev-unity3d** [comment]
> I would double check that you have merged master in correctly, adding right-click delete option to sticky notes shouldn't add 56k lines and remove 13k.

Oh crap, what I meant to do is to rebase my branch and then just push. But I did it through the client. I could revert that commit and rebase fully from the terminal

File: .yamato/upm-ci-hdrp.yml [hunk 4]
```diff
@@ -36,15 +36,27 @@ projects:
     image: {{ platform.image }}
     flavor: {{ platform.flavor}}
   commands:
-      - npm install upm-ci-utils -g --registry https://api.bintray.com/npm/unity/unity-npm
```
Comment (sophiaaar) [inline .yamato/upm-ci-hdrp.yml]: something seems weird with the merge - these yml files arent actually different from master, but theyre all showing as changed?

**Comment by alindmanUnity** [comment]
rebuilt here: #4534

**Pull request closed without merging**